gptcache/manager/scalar_data/sql_storage.py

```diff
@@ -0,0 +1,217 @@
+            )
+        else:
+            id = Column(Integer, primary_key=True, autoincrement=True)
+        question = Column(String(6000), nullable=False)
+        create_on = Column(DateTime, default=datetime.now)
+        last_access = Column(DateTime, default=datetime.now)
+        embedding_data = Column(LargeBinary, nullable=True)
+        deleted = Column(Integer, default=0)
+
+    class AnswerTable(Base):
+        """
+        answer table
+        """
+        __tablename__ = table_prefix + "_answer"
+        __table_args__ = {"extend_existing": True}
+
+        if db_type == "oracle":
+            id = Column(
+                Integer, Sequence("id_seq", start=1), primary_key=True, autoincrement=True
+            )
+        else:
+            id = Column(Integer, primary_key=True, autoincrement=True)
+        question_id = Column(Integer, nullable=False)
+        answer = Column(String(1000), nullable=False)
+        answer_type = Column(Integer, nullable=False)
+
+    return QuestionTable, AnswerTable
+
+
+class SQLStorage(CacheStorage):
+    """
+    Using sqlalchemy to manage SQLite, PostgreSQL, MySQL, MariaDB, SQL Server and Oracle.
+
+    :param name: the name of the cache storage, it is support 'sqlite', 'postgresql', 'mysql', 'mariadb', 'sqlserver' and  'oracle' now.
+    :type name: str
+    :param sql_url: the url of the sql database for cache, such as '<db_type>+<db_driver>://<username>:<password>@<host>:<port>/<database>',
+                    and the default value is related to the `cache_store` parameter, 'sqlite:///./sqlite.db' for 'sqlite',
+                    'postgresql+psycopg2://postgres:123456@127.0.0.1:5432/postgres' for 'postgresql',
+                    'mysql+pymysql://root:123456@127.0.0.1:3306/mysql' for 'mysql',
+                    'mariadb+pymysql://root:123456@127.0.0.1:3307/mysql' for 'mariadb',
+                    'mssql+pyodbc://sa:Strongpsw_123@127.0.0.1:1434/msdb?driver=ODBC+Driver+17+for+SQL+Server' for 'sqlserver',
+                    'oracle+cx_oracle://oracle:123456@127.0.0.1:1521/?service_name=helowin&encoding=UTF-8&nencoding=UTF-8' for 'oracle'.
+    :type sql_url: str
+    :param table_name: the table name for sql database, defaults to 'gptcache'.
+    :type table_name: str
+    """
+
+    def __init__(
+        self,
+        db_type: str = "sqlite",
+        url: str = "sqlite:///./sqlite.db",
+        table_name: str = "gptcache",
+    ):
+        self._url = url
+        self._ques, self._answer = get_models(table_name, db_type)
+        self._engine = create_engine(self._url)
+        self.Session = sessionmaker(bind=self._engine)  # pylint: disable=invalid-name
+        self._session = None
+        self.create()
+
+    def create(self):
+        self._ques.__table__.create(bind=self._engine, checkfirst=True)
+        self._answer.__table__.create(bind=self._engine, checkfirst=True)
+
+    def _insert(self, data: CacheData, session: "Session"):
```

Comment:
please use `Session`, not string, which is `sqlalchemy.orm.Session`

Reply:
fixed